### PR TITLE
Add mobile and phone number to register endpoint

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Check only Mobile phone and add it in mandatory fields.
Update also the OAS file.

File: src/server/rest/v1/service/AuthService.ts [hunk 1]
```diff
@@ -119,7 +120,7 @@ export default class AuthService {
 
   public static async handleRegisterUser(action: ServerAction, req: Request, res: Response, next: NextFunction): Promise<void> {
     // Filter
-    const filteredRequest = AuthValidator.getInstance().validateAuthSignOnReq(req.body);
+    const filteredRequest = AuthValidatorStorage.getInstance().validateAuthSignOn(req.body);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/AuthService.ts:123]: Why do you use AuthValidatorStorage, it's used for storing data in DB, use AuthValidator instead!

File: src/server/rest/v1/validator/AuthValidator.ts [hunk 2]
```diff
@@ -8,7 +8,6 @@ import global from '../../../../types/GlobalType';
 export default class AuthValidator extends SchemaValidator {
   private static instance: AuthValidator|null = null;
   private authSignIn: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/auth/auth-signin.json`, 'utf8'));
-  private authSignOn: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/auth/auth-signon.json`, 'utf8'));
```
Comment (LucasBrazi06) [inline src/server/rest/v1/validator/AuthValidator.ts:11]: Put everything back in AuthValidator class.

File: src/storage/mongodb/validator/AuthValidatorStorage.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { ChargingStationTemplate } from '../../../types/ChargingStation';
+import { HttpRegisterUserRequest } from '../../../types/requests/HttpUserRequest';
+import Schema from '../../../types/validator/Schema';
+import SchemaValidator from '../../../validator/SchemaValidator';
+import fs from 'fs';
+import global from '../../../types/GlobalType';
+
+export default class AuthValidatorStorage extends SchemaValidator {
```
Comment (LucasBrazi06) [inline src/storage/mongodb/validator/AuthValidatorStorage.ts:8]: Delete this class

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/server/rest/v1/validator/AuthValidator.ts (hunk 2 above)
Comment (mahdibenromdhane) [reply src/server/rest/v1/validator/AuthValidator.ts:11]: Done

File: src/server/rest/v1/validator/AuthValidator.ts [hunk 4]
```diff
@@ -16,7 +16,14 @@ export default class AuthValidator extends SchemaValidator {
   private authEula: Schema = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/server/rest/v1/schemas/auth/auth-eula.json`, 'utf8'));
 
   private constructor() {
-    super('AuthValidator');
+    super('AuthValidator', {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/validator/AuthValidator.ts:19]: There is no extra param in other validator, let the default one in the super class.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Check my comments + Have you made the mobile phone mandatory in Create/Update User front-end ?

**Comment by mahdibenromdhane** [comment]
> Check my comments + Have you made the mobile phone mandatory in Create/Update User front-end ?

Done now :)

**Review by LucasBrazi06 — APPROVED** [review]
You missed the entry in the OAS for the mobile
I disabled the mandatory mobile field the time for the mobile app to implement it also. To put in back once done.

**Pull request merged**